### PR TITLE
Add labels for ServiceMonitor K8S objects

### PR DESCRIPTION
**Pull request by csahithi**
#### What this PR does and why is it needed

This PR adds custom labels on ovn-kubernetes service monitor objects in the helm chart so that the service monitors are discovered from which the prometheus exporters scrape metrics off of.

#### Which issue(s) this PR fixes

None

#### Special notes for reviewers

None

#### Description for the changelog

Adds labels to service monitor ovn-kubernetes objects off of which prometheus metrics are scraped.

#### Does this PR introduce a user-facing change?

```release-note

```

### CODE REVIEW

**Comment by csahithi** [comment]
@girishmg Pls review this PR

File: helm/ovn-kubernetes/templates/ovnkube-monitor.yaml [hunk 1]
```diff
@@ -6,6 +6,9 @@ kind: ServiceMonitor
 metadata:
   labels:
     k8s-app: ovnkube-master
+    {{- with .Values.monitoring.serviceMonitorSelectorLabels }}
+    {{- toYaml . | nindent 4 }}
+    {{- end }}
```
Comment (girishmg) [inline helm/ovn-kubernetes/templates/ovnkube-monitor.yaml:11]: @csahithi can you please make the changes per our downstream code?

```
metadata:
  {{- with .Values.monitoring.serviceMonitorSelectorLabels }}
  labels:
    {{- toYaml . | nindent 4 }}
  {{- end }}
```

Comment (girishmg) [reply helm/ovn-kubernetes/templates/ovnkube-monitor.yaml:11]: move ` k8s-app: ovnkube-master` label to values.yaml along with the other label

**Review by girishmg — CHANGES_REQUESTED** [review]

File: helm/ovn-kubernetes/templates/ovnkube-monitor.yaml (hunk 1 above)
Comment (csahithi) [reply helm/ovn-kubernetes/templates/ovnkube-monitor.yaml:11]: @girishmg updated the value to `commonServiceMonitorSelectorLabels`

**Review by girishmg — APPROVED** [review]

**Comment by girishmg** [comment]
Some background:

1. When someone installs prometheus through prometheus operator, they can decide how to discover various serviceMonitors in the K8s clusters.
2. The option is set in `prometheus` CR object

In this example, all ServiceMonitors in all Namespaces are selected
```
# kubectl get prometheus -n monitoring -o yaml |grep -i servicemonitor
    serviceMonitorNamespaceSelector: {}
    serviceMonitorSelector: {}
# 
```

In this example, only those servicemontiors with the label `release: kube-prometheus-stack` will be selected

```
         serviceMonitorNamespaceSelector: {}
         serviceMonitorSelector:
             matchLabels:
                   release: kube-prometheus-stack
```

**Comment by girishmg** [comment]
Changes were only made to Helm Charts, and the helm lane test ran successfully.. so going to merge this.

**Pull request merged**